src/Features/Core/Portable/LegacySolutionEvents/ILegacyWorkspaceDescriptor.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.LegacySolutionEvents
+{
+    internal interface ILegacyWorkspaceDescriptor
```

Comment:
>ILegacyWorkspaceDescriptor [](http://example.com/codeflow?start=23&length=26)

Maybe -Accessor would be better name?

Reply:
Isn't WorkspaceKind available from Solution as well?

Reply:
and SolutionServices also. so perhaps this just needs CurrentSolution?

Reply:
WFM.  Will rename in followup.

Reply:
> Isn't WorkspaceKind available from Solution as well?
> and SolutionServices also. so perhaps this just needs CurrentSolution?

True!  Ok, i can simplify this even further.  Will do so in next PR :)